src/Builder/DownloadFromTrait.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace Sensiolabs\GotenbergBundle\Builder;
+
+use Sensiolabs\GotenbergBundle\Exception\MissingRequiredFieldException;
+use Symfony\Component\Mime\Part\DataPart;
+
+trait DownloadFromTrait
+{
+    /**
+     * Sets download from to download each entry (file) in parallel (default None).
+     * (URLs MUST return a Content-Disposition header with a filename parameter.).
+     *
+     * @see https://gotenberg.dev/docs/routes#download-from
+     *
+     * @param list<array{url: string, extraHttpHeaders?: array<string, string>}> $downloadFrom
+     */
+    abstract public function downloadFrom(array $downloadFrom): static;
+
+    /**
+     * @param array{downloadFrom?: array<string, array{url: string, extraHttpHeaders?: array<string, string>}>} $formFields
+     * @param list<array{url: string, extraHttpHeaders?: array<string, string>}>                                $downloadFrom
+     */
+    private function withDownloadFrom(array &$formFields, array $downloadFrom): static
+    {
+        if ([] === $downloadFrom) {
+            unset($formFields['downloadFrom']);
+
+            return $this;
+        }
+
+        $formFields['downloadFrom'] = [];
+
+        foreach ($downloadFrom as $file) {
+            if (!\array_key_exists('url', $file)) {
+                throw new MissingRequiredFieldException('Missing field "url"');
+            }
+
+            $formFields['downloadFrom'][$file['url']] = $file;
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param array<string, array{url: string, extraHttpHeaders?: array<string, string>}> $value
+     *
+     * @return array<string, array<string|int, mixed>|string|\Stringable|int|float|bool|\BackedEnum|DataPart>
+     */
+    private function downloadFromNormalizer(array $value, callable $encoder): array
+    {
+        $downloadsFrom = array_values($value);
+        $data = [];
+
+        array_push($data, ...$downloadsFrom);
+
+        return $encoder('downloadFrom', $data);
```

Comment:
Why this `array_push`? https://3v4l.org/3XaHH
```suggestion
        return $encoder('downloadFrom', array_values($value));
```

Reply:
No idea why I did that :exploding_head:

Reply:
![need vacation](https://i.giphy.com/media/v1.Y2lkPTc5MGI3NjExZzdzYWpoNXA5OXdvc3dpZDN2bjc4eGVzYnU4ejAxdWw2a3lxeHpvYiZlcD12MV9pbnRlcm5hbF9naWZfYnlfaWQmY3Q9Zw/lJEGgG5ajs4zC/giphy.gif)